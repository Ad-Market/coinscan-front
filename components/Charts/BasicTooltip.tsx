import type { TooltipProps } from 'recharts';
import Box from '@mui/material/Box';

import { formatValue } from '../../utils/formatters';

const BasicTooltip = ({
  active,
  payload,
  label,
  labelFormatter,
}: TooltipProps<string, number>) => {
  if (active && payload && payload.length > 0) {
    return (
      <Box
        sx={{
          translate: ['0 -40px', '0 0'],
          p: [0.5, 2],
          fontSize: ['0.75rem', 'body2.fontSize'],
          color: 'white',
          backgroundColor: 'grey.500',
        }}
      >
        {labelFormatter ? labelFormatter(label, payload) : label}:{' '}
        {formatValue(payload[0].payload.display || 0)}
      </Box>
    );
  }
  return null;
};

export default BasicTooltip;
