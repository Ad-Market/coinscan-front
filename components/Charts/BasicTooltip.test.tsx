import { render } from '@testing-library/react';

import {
  CHART_TIME_TOOLTIP_LABEL_FORMATTERS,
  ChartTimeGroupings,
} from '../../utils/charts';
import BasicTooltip from './BasicTooltip';

test('HoldersTooltip renders correctly for daily grouping', () => {
  const payload = [
    {
      payload: { count: 6293, name: '1638144000' },
    },
  ];

  const { container } = render(
    <BasicTooltip
      active
      label={1654473600}
      payload={payload}
      labelFormatter={
        CHART_TIME_TOOLTIP_LABEL_FORMATTERS[ChartTimeGroupings.BY_DAY]
      }
    />
  );

  expect(container).toMatchSnapshot('rendered');
});

test('HoldersTooltip renders correctly for weekly grouping', () => {
  const payload = [
    {
      payload: { count: 5393, name: '1638144000' },
    },
  ];

  const { container } = render(
    <BasicTooltip
      active
      label={payload[0].payload.name}
      payload={payload}
      labelFormatter={
        CHART_TIME_TOOLTIP_LABEL_FORMATTERS[ChartTimeGroupings.BY_WEEK]
      }
    />
  );

  expect(container).toMatchSnapshot('rendered');
});

test('HoldersTooltip renders correctly for monthly grouping', () => {
  const payload = [
    {
      payload: {
        count: 5273,
        name: '1635724800',
      },
    },
  ];

  const { container } = render(
    <BasicTooltip
      active
      label={payload[0].payload.name}
      payload={payload}
      labelFormatter={
        CHART_TIME_TOOLTIP_LABEL_FORMATTERS[ChartTimeGroupings.BY_MONTH]
      }
    />
  );

  expect(container).toMatchSnapshot('rendered');
});
