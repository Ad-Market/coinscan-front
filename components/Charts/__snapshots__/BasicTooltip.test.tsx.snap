// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceChartTooltip renders correctly for daily grouping: rendered 1`] = `
<div>
  <div
    class="MuiBox-root css-f4glh7"
  >
    11/1/2021
    :
     
    1,234,567,891.1234568
  </div>
</div>
`;
